### PR TITLE
Agent: engage in ongoing challenges only at unfilled requests

### PR DESCRIPTION
**Pull request by fredo**
fixes: #571 

tests are missing yet

### CODE REVIEW

File: beamer/chain.py [hunk 1]
```diff
@@ -382,19 +382,29 @@ def maybe_challenge(claim: Claim, context: Context) -> bool:
...
 
-    if claim.is_challenger_winning and claim.claimer != context.address:
-        return False
-
     agent_winning = context.address in claim.get_winning_addresses()
     if agent_winning:
         return False
 
+    own_challenge_stake = Wei(claim.get_challenger_stake(context.address))
+    agent_losing = claim.claimer == context.address or own_challenge_stake > 0
```
Comment (palango) [inline beamer/chain.py:390]: maybe we can find a better name for the variable

Comment (palango) [reply beamer/chain.py:390]: I think `agent_participating` fits better

Comment (palango) [reply beamer/chain.py:390]: Yes, I just feel this makes it harder to read the condition below. But it's just a nitpick, both are fine.

File: beamer/tests/agent/unit/test_challenge.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+def test_challenge_as_claimer():
+    context, _ = make_context()
+
+    request = make_request()
+    context.requests.add(request.id, request)
+    request.filler = make_address()
+    claim = make_claim(
+        request=request,
+        claimer=context.address,
+        challenger=make_address(),
```
Comment (palango) [inline beamer/tests/agent/unit/test_challenge.py:48]: you don't strictly need this, it will be set automatically.

**Review by palango — APPROVED** [review]
Looks good, and nice tests!

File: beamer/tests/agent/unit/test_challenge.py (hunk 2 above)
Comment (fredo) [reply beamer/tests/agent/unit/test_challenge.py:49]: for claimer you are right, but I think for challenger it's not the case (can be None)

Comment (fredo) [reply beamer/tests/agent/unit/test_challenge.py:50]: yeah just came across that .. I think, I will change that in order to be able to create an unchallenged claim.

**Pull request merged**